src/Route.php

```diff
@@ -2,56 +2,36 @@
 
 namespace Yiisoft\Router;
 
+use InvalidArgumentException;
+use LogicException;
 use Yiisoft\Router\Middleware\Callback;
 use Psr\Http\Message\ResponseInterface;
 use Psr\Http\Message\ServerRequestInterface;
 use Psr\Http\Server\MiddlewareInterface;
 use Psr\Http\Server\RequestHandlerInterface;
 
+use function current;
+use function is_callable;
+use function next;
+use function reset;
+
 /**
  * Route defines a mapping from URL to callback / name and vice versa
  */
 final class Route implements MiddlewareInterface, RequestHandlerInterface
 {
+    private string $name;
```

Comment:
We need to initialize it since it's optional.

```suggestion
    private ?string $name = null;
```